src/Extractor.cc

```diff
@@ -2168,14 +2168,53 @@ namespace insur {
 	ril.index = discNumber;
 
 
-	//if (zmin > 0) {	
+	//if (zmin > 0) 	
         std::ostringstream dname, pconverter;
 	//disk name
         dname << xml_disc << discNumber; // e.g. Disc6
 
         std::map<int, ERingInfo> rinfo;
 	std::set<int> ridx;
 
+      
```

Comment:
Yes I had a glance at your presentation's slides last Friday.
The Ring algo is used to place modules within a ring, in, for example, the forward section, in order to avoid lengthy XMLs.
Basically, an other option would have been to place each module individually within a ring (ie, numModulesPerRing XML Pospart sections).
Since the ring was regular, there was no point doing so.

Now, since you need to have specfic position for each module within a ring, there is no point using the algorithm. It will not work without modifying the algo in CMSSW, with risks of regressions in all the numerous placed it is used, and additonnally has no advantage, because there is no regularity that needs to be abstracted into an algo here.

When isRegularRing() is wrong, you could just remove the algo section, and instead add numModulesPerRing sections, adding each single module position within the Ring.
You could simply describe these PosPart placements with one translation and one rotation (which is exactly what you need).